WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {
+  Button, ButtonGroup, Confirm, Form, Header, List, Modal,
+} from 'semantic-ui-react';
+import useLoadedData from '../../../lib/hooks/useLoadedData';
+import { fetchUserGroupsUrl, addUserGroupsUrl, userGroupsUpdateUrl } from '../../../lib/requests/routes.js.erb';
+import Errored from '../../Requests/Errored';
+import Loading from '../../Requests/Loading';
+import useSaveAction from '../../../lib/hooks/useSaveAction';
+
+const defaultRegion = {
+  name: '',
+  group_type: 'delegate_regions',
+  parent_group_id: null,
+  is_active: true,
+  is_hidden: false,
+};
+
+function UserGroupVisibility({ userGroup, save, sync }) {
+  const [open, setOpen] = React.useState(false);
+  const iconName = userGroup.is_active ? 'eye' : 'eye slash';
+  return (
+    <>
+      <List.Icon
+        name={iconName}
+        link
+        onClick={() => setOpen(true)}
+      />
+      <Confirm
+        open={open}
+        content={`Are you sure you want to ${userGroup.is_active ? 'deactivate' : 'activate'} ${userGroup.name}?`}
+        onCancel={() => setOpen(false)}
+        onConfirm={() => {
+          setOpen(false);
+          save(
+            userGroupsUpdateUrl(userGroup.id),
+            { is_active: !userGroup.is_active, is_hidden: false },
+            sync,
+          );
+        }}
+      />
+    </>
+  );
+}
+
+export default function RegionManager() {
+  const {
+    data, loading, error, sync,
+  } = useLoadedData(fetchUserGroupsUrl('delegate_regions'));
+  const { save, saving } = useSaveAction();
+  const [openModalType, setOpenModalType] = React.useState();
+  const [newRegion, setNewRegion] = React.useState(defaultRegion);
+  const [saveError, setSaveError] = React.useState();
+
+  const closeModal = () => setOpenModalType(null);
+
+  const regions = React.useMemo(() => data?.filter((group) => !group.parent_group_id).sort(
+    (group1, group2) => group1.name.localeCompare(group2.name),
+  ), [data]);
+
+  const subRegions = React.useMemo(() => {
+    const subRegionsMap = {};
+    data?.forEach((group) => {
+      if (group.parent_group_id) {
+        if (!subRegionsMap[group.parent_group_id]) {
+          subRegionsMap[group.parent_group_id] = [];
+        }
+        subRegionsMap[group.parent_group_id].push(group);
+      }
+    });
+    return subRegionsMap;
```

Comment:
```suggestion
    const subRegions = data?.filter((group) => group.parent_group_id);
    return Object.groupBy(subRegions, (group) => group.parent_group_id);
```

Reply:
cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

Reply:
New learning again, thank you.